ЛР 3 Вар 4 (Порушенков) ВТ-119

Vitaliy308 opened the pull request:
Не реализована первая часть индивидуального задания (реализация преобразования множества элементов в сбалансированное бинарное дерево)

GOOD-Stuff commented on lab_04/src/main.cpp:
```diff
@@ -0,0 +1,64 @@
+
+    try{
+        fstream fin;
+        fin.open(R"(C:\Users\User\Desktop\ssw_lab_4sem\lab_04\src\name.txt)");
```
Старайтесь не использовать абсолютные пути до файлов (по крайне мере в коде, к-ый потом публикуете).

GOOD-Stuff commented on lab_04/src/main.cpp:
```diff
@@ -0,0 +1,64 @@
+#include <string>
+#include "map.h"
+
+using namespace std;
```
На будущее, лучше не используйте полное включение содержимого `std`.
Если, например, подключить сюда STL *map*, то можно получить очень нехороший эффект. 
Вам зачастую из *std* нужна лишь небольшая часть функционала (например `cout/cerr/cin`), в таком случае лучше включать именно её:
```
using std::cout;
using std::cerr;
using std::cin;
```

Vitaliy308 committed:
completing

Не реализована первая часть индивидуального задания (реализация преобразования множества элементов в сбалансированное бинарное дерево)

GOOD-Stuff commented:
Всегда в названии своих реквестов указывайте:
- группу
- номер работы
- вариант
- фамилии участников

иначе как мне понять, кто вы и что вы сдаёте?!

Vitaliy308 commented:
- ВТ-119
- ЛР 3
- Вариант 4
- Порушенков Виталий

Нет, скорее всего не будут реализовываться

GOOD-Stuff commented on src/Binary_tree.h:
```diff
@@ -0,0 +1,345 @@
+
+template<class T>
+std::vector<bin_tree<T>*> bin_tree<T>::split() {
+	std::vector<bin_tree<T>*> result{ new bin_tree<T>(),new bin_tree<T>() };
```
Не уверен, что было хорошей идеей отдельно выделять деревья в куче, когда vector и так память выделяет в дин. области. 
Вроде где-то здесь санитайзер мне и показывает утечку.

GOOD-Stuff reviewed (commented):
За работу ставлю 8
